osutil/user.go

```diff
@@ -170,3 +170,28 @@ func UidGid(u *user.User) (sys.UserID, sys.GroupID, error) {
 
 	return sys.UserID(uid), sys.GroupID(gid), nil
 }
+
+func UserConfig(usr *user.User, filename string) string {
+	//  phase 0: pretend XDG_CONFIG_HOME is not a thing
+	//  phase 1: check for XDG_CONFIG_HOME/snap, only use it if it exists
+	//  phase 2: check for ~/.snap, only use it if it exists
```

Comment:
Is phase 2 something you plan on doing later? Phase 1 sounds just like what we want. Are phases fallbacks?

Reply:
phase 2 is probably needed before we think about (allowing) moving `~/snap` to `~/.snap`